main_test.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+)
+
+func TestToBase10(t *testing.T) {
+	for _, tc := range []struct {
+		name       string
+		number     string
+		base       int
+		wantNumber int
+	}{
+		{"empty", "", 16, 0},
+		{"from base 10 to base 10", "123", 10, 123},
+		{"base 2 to base 10", "111101", 2, 61},
+		{"base 16 to base 10", "ABCD", 16, 43981},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			if got := toBase10(tc.number, tc.base); got != tc.wantNumber {
+				t.Errorf("toBase10(num = %v, fromBase = %v) = %v, want %v", tc.number, tc.base, got, tc.wantNumber)
+			}
+		})
+	}
+}
+
+func TestFromBase10(t *testing.T) {
+	for _, tc := range []struct {
+		name       string
+		number     int
+		base       int
+		wantNumber string
+	}{
+		{"zero", 0, 16, "0"},
```

Comment:
Try the following test: `number = 2`, `base = 2`. Your program outputs `2`. This is not correct.